### PR TITLE
slim docker images

### LINKED ISSUE #3527: 建议压缩docker镜像大小
**建议压缩docker镜像大小**

目前的镜像有点大

建议分两阶段编译，以压缩镜像大小

以下举例说明

```

FROM openjdk:8-jre-alpine

RUN  apk update upgrade \
    && apk add --no-cache unzip 

ADD apollo-configservice-${VERSION}-github.zip /apollo-configservice/apollo-configservice-${VERSION}-github.zip

RUN unzip /apollo-configservice/apollo-configservice-${VERSION}-github.zip -d /apollo-configservice \
    && rm -rf /apollo-configservice/apollo-configservice-${VERSION}-github.zip \
    && chmod +x /apollo-configservice/scripts/startup.sh

FROM openjdk:8-jre-alpine

COPY --from=0 /apollo-configservice /apollo-configservice

# 其他操作继续

```

### PR DESCRIPTION
**Pull request by finchcn**
## What's the purpose of this PR

压缩docker镜像大小

## Which issue(s) this PR fixes:
Fixes #3527

## Brief changelog

压缩docker镜像大小

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Read the [Contributing Guide](https://github.com/ctripcorp/apollo/blob/master/CONTRIBUTING.md) before making this pull request.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit tests to verify the code.
- [x] Run `mvn clean test` to make sure this pull request doesn't break anything.

### CODE REVIEW

File: apollo-adminservice/src/main/docker/Dockerfile [hunk 1]
```diff
@@ -4,24 +4,36 @@
 # 3. Run with: docker run -p 8090:8090 -e SPRING_DATASOURCE_URL="jdbc:mysql://fill-in-the-correct-server:3306/ApolloConfigDB?characterEncoding=utf8" -e SPRING_DATASOURCE_USERNAME=FillInCorrectUser -e SPRING_DATASOURCE_PASSWORD=FillInCorrectPassword -d -v /tmp/logs:/opt/logs --name apollo-adminservice apolloconfig/apollo-adminservice
 
 FROM openjdk:8-jre-alpine
-MAINTAINER ameizi <sxyx2008@163.com>
+LABEL maintainer="finchcn@gmail.com;ameizi<sxyx2008@163.com>"
+
+RUN echo "http://mirrors.aliyun.com/alpine/v3.8/main" > /etc/apk/repositories \
+    && echo "http://mirrors.aliyun.com/alpine/v3.8/community" >> /etc/apk/repositories \
+    && apk update upgrade \
+    && apk add --no-cache unzip \
```
Comment (nobodyiam) [inline apollo-adminservice/src/main/docker/Dockerfile:12]: I think we need to remove the trailer `\`?
```suggestion
    && apk add --no-cache unzip
```

**Review by nobodyiam — CHANGES_REQUESTED** [review]
It doesn't work in my environment..

mvn docker:build -pl apollo-adminservice

```
[INFO] Scanning for projects...
[INFO]
[INFO] -----------< com.ctrip.framework.apollo:apollo-adminservice >-----------
[INFO] Building Apollo AdminService 1.8.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- docker-maven-plugin:1.2.2:build (default-cli) @ apollo-adminservice ---
[INFO] Using authentication suppliers: [ConfigFileRegistryAuthSupplier, FixedRegistryAuthSupplier]
[INFO] Copying /Users/jason/git/mine/apollo/apollo-adminservice/target/apollo-adminservice-1.8.0-SNAPSHOT-github.zip -> /Users/jason/git/mine/apollo/apollo-adminservice/target/docker/apollo-adminservice-1.8.0-SNAPSHOT-github.zip
[INFO] Copying /Users/jason/git/mine/apollo/apollo-adminservice/src/main/docker/Dockerfile -> /Users/jason/git/mine/apollo/apollo-adminservice/target/docker/Dockerfile
[INFO] Building image apolloconfig/apollo-adminservice
[WARNING]: Empty continuation line found in:
    RUN echo "http://mirrors.aliyun.com/alpine/v3.8/main" > /etc/apk/repositories     && echo "http://mirrors.aliyun.com/alpine/v3.8/community" >> /etc/apk/repositories     && apk update upgrade     && apk add --no-cache unzip ARG VERSION
[WARNING]: Empty continuation lines will become errors in a future release.

Step 1/14 : FROM openjdk:8-jre-alpine

 ---> f7a292bbb70c
Step 2/14 : LABEL maintainer="finchcn@gmail.com;ameizi<sxyx2008@163.com>"

 ---> Using cache
 ---> 9414208b4f9a
Step 3/14 : RUN echo "http://mirrors.aliyun.com/alpine/v3.8/main" > /etc/apk/repositories     && echo "http://mirrors.aliyun.com/alpine/v3.8/community" >> /etc/apk/repositories     && apk update upgrade     && apk add --no-cache unzip ARG VERSION

 ---> Running in ca5cc9b2bf1a
fetch http://mirrors.aliyun.com/alpine/v3.8/main/x86_64/APKINDEX.tar.gz
fetch http://mirrors.aliyun.com/alpine/v3.8/community/x86_64/APKINDEX.tar.gz
v3.8.5-65-g097f0358a9 [http://mirrors.aliyun.com/alpine/v3.8/main]
v3.8.5-37-gf06ffe835a [http://mirrors.aliyun.com/alpine/v3.8/community]
OK: 9616 distinct packages available
fetch http://mirrors.aliyun.com/alpine/v3.8/main/x86_64/APKINDEX.tar.gz
fetch http://mirrors.aliyun.com/alpine/v3.8/community/x86_64/APKINDEX.tar.gz
ERROR: unsatisfiable constraints:

  ARG (missing):
    required by: world[ARG]
  VERSION (missing):
    required by: world[VERSION]
```

File: apollo-adminservice/src/main/docker/Dockerfile (hunk 1 above)
Comment (finchcn) [reply apollo-adminservice/src/main/docker/Dockerfile]: fixed

Comment (nobodyiam) [reply apollo-adminservice/src/main/docker/Dockerfile]: Thanks, please also help to fix the conflicts.

image

Comment (finchcn) [reply apollo-adminservice/src/main/docker/Dockerfile]: I had a little problem when trying to fix the conflicts. I refork the repository, and resoved the conflicts, but there is no push notification.
The status in tab of "File Changed" is OK.
Need I create a new PR?

Comment (nobodyiam) [reply apollo-adminservice/src/main/docker/Dockerfile]: Maybe there is something wrong duing the conflict resolving process, it's ok to create a new pr.

**Comment by finchcn** [comment]
recreate PR #3572

**Pull request closed without merging**